fdbclient/common.go

```diff
@@ -73,6 +73,50 @@ func getFDBDatabase(cluster *fdbv1beta2.FoundationDBCluster) (fdb.Database, erro
 	return database, nil
 }
 
+// getConnectionStringFromDB gets the database's connection directly from the system key
+func getConnectionStringFromDB(cluster *fdbv1beta2.FoundationDBCluster) (string, error) {
+	fdbKey := "\xff\xff/connection_string"
+
+	database, err := getFDBDatabase(cluster)
+	if err != nil {
+		return "", err
+	}
+
+	result, err := database.Transact(func(transaction fdb.Transaction) (interface{}, error) {
+		err := transaction.Options().SetAccessSystemKeys()
+		if err != nil {
+			return nil, err
+		}
+		// Wait default timeout seconds to receive status for larger clusters.
+		err = transaction.Options().SetTimeout(int64(DefaultCLITimeout * 1000))
```

Comment:
We wait here longer than the default timeout. The comment should be adjusted, we are not reading the cluster status.